### PR TITLE
MueLu: refactore of aggregation phase 1, see issue #5838

### PR DESCRIPTION
**Pull request by lucbv**
@trilinos/muelu

## Description
Two new aggregation algorithms are implemented:
    1) is a direct refactor of the serial algorithm using kokkos
    2) is a similar approach to 1) but provides deterministic aggregates

## Motivation and Context
This refactor work will allow MueLu's setup to be performed on GPU which is critical to application that need to run at scale on NGP architectures

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of #5838
* Composed of 

## How Has This Been Tested?
A local build has been performed and test pass, a new unit-test  has been added to test the deterministic algorithms.

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

File: packages/muelu/src/Graph/UncoupledAggregation/MueLu_UncoupledAggregationFactory_kokkos_def.hpp [hunk 1]
```diff
@@ -184,26 +188,40 @@ namespace MueLu {
...
+    // getLocalMap to have a Kokkos::View on the appropriate memory_space
+    // instead of an ArrayRCP.
+    // // TODO
+    // //ArrayRCP<const bool> dirichletBoundaryMap = graph->GetBoundaryNodeMap();
+    // ArrayRCP<const bool> dirichletBoundaryMap;
+
+    // if (dirichletBoundaryMap != Teuchos::null)
+    //   for (LO i = 0; i < numRows; i++)
+    //     if (dirichletBoundaryMap[i] == true)
+    //       aggStat[i] = BOUNDARY;
```
Comment (jhux2) [inline packages/muelu/src/Graph/UncoupledAggregation/MueLu_UncoupledAggregationFactory_kokkos_def.hpp]: Is there value in preserving this commented-out code?

File: packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase1Algorithm_kokkos_def.hpp [hunk 2]
```diff
@@ -109,19 +100,35 @@ namespace MueLu {
     //can only enforce max aggregate size
     if(algorithm == Algorithm::Distance2)
     {
-      BuildAggregatesDistance2(graph, aggregates, aggStat,
-                               numNonAggregatedNodes, maxNodesPerAggregate);
+      if(params.get<bool>("aggregation: deterministic"))
+      {
+        std::cout << "Using deterministic aggregation for phase 1" << std::endl;
```
Comment (jhux2) [inline packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase1Algorithm_kokkos_def.hpp]: Every process is going to print ....

**Comment by lucbv** [comment]
I think it's time for a build on white!

File: packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase1Algorithm_kokkos_def.hpp (hunk 2 above)
Comment (lucbv) [reply packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase1Algorithm_kokkos_def.hpp]: Yep, that was a debug statement that is now removed.

**Comment by lucbv** [comment]
@framework apparently something is going wrong with the `intel-17.0.1` license on [ascic158](https://testing.sandia.gov/cdash/viewConfigure.php?buildid=5610499)
Thanks for taking a look at it.

**Pull request merged**